app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -253,6 +253,7 @@ def facts
         sync_task(::Actions::Katello::Host::Update, @host, rhsm_params.except(:facts))
         Katello::Host::SubscriptionFacet.update_facts(@host, rhsm_params[:facts]) unless rhsm_params[:facts].nil?
         update_host_registered_through(@host, request.headers)
+        @host.subscription_facet.candlepin_consumer.update(rhsm_params.slice(:facts))
```

Comment:
Should we also avoid updating the facts here if they are nil similar to line 254?